src/main/python/inverse_parser/_merged_types_restorer.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Aniskov N.
+
+import ast
+
+
+_py_node_type_name_to_ast_type = \
+    dict([(name, cls) for name, cls in ast.__dict__.items() if isinstance(cls, type)])
+
+
+class _MergedTypesRestorer:
+    """
+    Class provides functionality to restore context and operations,
+    encoded in XML nodes tags
+    """
+
+    @staticmethod
+    def get_types_from_tag(xml_node):
+        """
+        :param xml_node: node of AST in XML format
+        :return py_node_type: type of corresponding Python AST node
+        :return merged_types: other types merged in tag: context or operators
+        """
+        types = xml_node.tag.split('_')
+        py_node_type_name, merged_types_names = types[0], types[1:]
+        if py_node_type_name == 'NoneType':
+            py_node_type = type(None)
+        else:
+            py_node_type = _py_node_type_name_to_ast_type[py_node_type_name]
+        merged_types = [_py_node_type_name_to_ast_type[merged_type_name]
+                        for merged_type_name in merged_types_names]
+        return py_node_type, merged_types
+
+    @staticmethod
+    def set_ctx_and_ops(py_node, merged_types):
+        """
+        :param py_node: Python ast tree node
+        :param merged_types: type of operators or context extracted from XML tag
+        :return:
```

Comment:
Should we add return type annotation?
`def set_ctx_and_ops(py_node, merged_types): -> None`

Reply:
Yes, sure! Thanks!